.github/workflows/build-pack-release.yml

```
@@ -0,0 +1,96 @@
+name: Build-Pack-Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: The version to tag the release with, e.g., 1.2.0, 1.3.0
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+    build-pack-release:
+        runs-on: ${{ matrix.os }}
+
+        env:
+            DOTNET_MULTILEVEL_LOOKUP: 1
+
+        strategy:
+            fail-fast: true
+            matrix:
+                os: [windows-latest]
+
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+
+        - name: Configure AWS credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_RELEASE }}
+            aws-region: us-west-2
+
+        - name: Get Strong-Naming Key
+          run: aws s3api get-object --bucket ${{ secrets.SNK_BUCKET }} --key awsxrayrecorder.dll.snk awsxrayrecorder.dll.snk
+          
+        - name: Install dependencies
+          run: dotnet restore .\sdk\AWSXRayRecorder.sln
+
+        - name: Build solution
+          run: dotnet build .\sdk\AWSXRayRecorder.sln /p:Configuration=Release --no-restore /p:AssemblyOriginatorKeyFile=${{ github.workspace }}\awsxrayrecorder.dll.snk
+
+        - name: Assume signer role
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            role-to-assume: ${{ secrets.AWS_ARTIFACT_ACCESS_ROLE_ARN }}
+            aws-region: us-west-2
+
+        - name: Invoke Signing script
+          run: |
+            .\buildtools\sign_files.ps1 -Path .\sdk\src -Filter AWSXRayRecorder.*.dll -Recurse
+          env:
+            UNSIGNED_BUCKET: ${{ secrets.AWS_UNSIGNED_BUCKET_NAME }}
+            SIGNED_BUCKET: ${{ secrets.AWS_SIGNED_BUCKET_NAME }}
+
+        - name: Pack nugets
+          run: |
+            dotnet pack .\sdk\AWSXRayRecorder.sln  --no-build -c Release -o .\Deployment\nuget-packages
+            nuget pack .\sdk\AWSXRayRecorder.nuspec -OutputDirectory .\Deployment\nuget-packages -Exclude **
+
+        - name: Upload nugets as actions artifact
```

At first I was confused why we weren't then downloading the artifact, but I remember there was value to uploading it so we can see what was uploaded on the action itself.

nit:
```suggestion
        - name: Upload nugets to this GitHub Action run as an artifact
```